TestingApp/VisitorInfo/VisitorInfoViewController.HeaderView.swift

```diff
@@ -0,0 +1,171 @@
+import UIKit
+
+final class HeaderView: UITableViewHeaderFooterView {
+    static var identifier: String { "\(Self.self)" }
+    static var height: CGFloat { UITableView.automaticDimension }
+
+    private lazy var titleLabel = UILabel()
+    private lazy var segmentedControl = UISegmentedControl()
+    private lazy var actionButton = UIButton(type: .system)
+
+    var props = Props(title: "", segments: [], selectedSegment: nil, actionButton: nil, segmentAccessibilityIdentifierPrefix: "") {
+        didSet {
+            guard props != oldValue else { return }
+            renderProps()
+        }
+    }
+
+    override init(reuseIdentifier: String?) {
+        super.init(reuseIdentifier: reuseIdentifier)
+        setup()
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) { fatalError("init(coder:) has not been implemented") }
+
+    private func setup() {
+        let margins = 10.0
+
+        contentView.addSubview(titleLabel)
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        titleLabel.font = .systemFont(ofSize: 18, weight: .semibold)
+
+        let stackView = UIStackView()
+        stackView.spacing = margins
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        contentView.addSubview(stackView)
+
+        stackView.addArrangedSubview(segmentedControl)
+        segmentedControl.addTarget(
+            self,
+            action: #selector(handleSegmentedControlSelect),
+            for: .valueChanged
+        )
+
+        stackView.addArrangedSubview(actionButton)
+        actionButton.addTarget(
+            self,
+            action: #selector(handleActionButtonTap),
+            for: .touchUpInside
+        )
+        actionButton.titleLabel?.textAlignment = .center
+        actionButton.titleLabel?.font = .monospacedSystemFont(ofSize: 30, weight: .heavy)
+
+        NSLayoutConstraint.activate([
+            titleLabel.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: margins),
+            titleLabel.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -margins),
+            titleLabel.topAnchor.constraint(equalTo: contentView.topAnchor, constant: margins),
+            titleLabel.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -margins),
+            stackView.centerYAnchor.constraint(equalTo: contentView.centerYAnchor),
+            stackView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -margins),
+        ])
+    }
+
+    func renderProps() {
+        titleLabel.text = props.title
+        renderedSegments = .init(segments: props.segments, selectedSegment: props.selectedSegment) //props.segments
```

Comment:
Is there a reason for this comment?

Reply:
No, removed b9b42dcdc7b51038e214ce28b0e22876220c5bf0.